src/rail/estimation/algos/random_forest.py

```diff
@@ -0,0 +1,166 @@
+    outputs = [('model', ModelHandle)]
+    
+    def __init__(self, args, comm=None):
+        CatInformer.__init__(self, args, comm=comm)
+        
+    def run(self):
+        # Load the training data
+        if self.config.hdf5_groupname:
+            training_data_table = self.get_data('input')[self.config.hdf5_groupname]
+        else:  # pragma: no cover
+            training_data_table = self.get_data('input')
+
+        # Pull out the appropriate columns and combinations of the data
+        print(f"Using these bands to train the tomography selector: {self.config.bands}")
+        
+        # Generate the training data that we will use
+        # We record both the name of the column and the data itself
+        features = []
+        training_data = []
+        for b1 in self.config.class_bands[:]:
+            b1_cat=self.config.bands[b1]
+            # First we use the magnitudes themselves
+            features.append(b1)
+            training_data.append(training_data_table[b1_cat])
+            # We also use the colours as training data, even the redundant ones
+            for b2 in self.config.class_bands[:]:
+                b2_cat=self.config.bands[b2]
+                if b1 < b2:
+                    features.append(f"{b1}-{b2}")
+                    training_data.append(training_data_table[b1_cat] - training_data_table[b2_cat])
+        training_data = np.array(training_data).T
+
+        print("Training data for bin classifier has shape ", training_data.shape)
+
+        # Now put the training data into redshift bins
+        # We use -99 to indicate that we are outside the desired ranges
+        z = training_data_table[self.config.redshift_col]
+        training_bin = np.repeat(self.config.no_assign, len(z))
+        print("Using these bin edges:", self.config.bin_edges)
+        for i, zmin in enumerate(self.config.bin_edges[:-1]):
+            zmax = self.config.bin_edges[i + 1]
+            training_bin[(z > zmin) & (z < zmax)] = i
+            ntrain_bin = ((z > zmin) & (z < zmax)).sum()
+            print(f"Training set: {ntrain_bin} objects in tomographic bin {i}")
+
+        # Can be replaced with any classifier
+        skl_classifier = skl_RandomForestClassifier(
+            max_depth=10,
+            max_features=None,
+            n_estimators=20,
+            random_state=self.config.random_seed,
+        )
+        skl_classifier.fit(training_data, training_bin)
+
+        #return classifier, features
+        self.model = randomForestmodel(skl_classifier, features)
+        self.add_data('model', self.model)
+        
+
+class randomForestClassifier(CatClassifier):
+    """Classifier that assigns tomographic 
+    bins based on random forest method"""
+    
+    name = 'randomForestClassifier'
+    config_options = CatClassifier.config_options.copy()
+    config_options.update(
+        id_name=Param(str, "", msg="Column name for the object ID in the input data, if empty the row index is used as the ID."),
```

Comment:
Nothing needs to be changed here, but I just want to note re: LSSTDESC/rail_base#39 that this will be the first instance of IDs in the RAIL-iverse so should be considered in decision-making about how to consistently reference these throughout `src/rail` code.